rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -1757,26 +1766,16 @@ extern "C" rmw_node_t * rmw_create_node(
   RET_ALLOC_X(node->namespace_, return nullptr);
   memcpy(const_cast<char *>(node->namespace_), namespace_, strlen(namespace_) + 1);
 
-  {
-    // Though graph_cache methods are thread safe, both cache update and publishing have to also
-    // be atomic.
-    // If not, the following race condition is possible:
-    // node1-update-get-message / node2-update-get-message / node2-publish / node1-publish
-    // In that case, the last message published is not accurate.
-    auto common = &context->impl->common;
-    std::lock_guard<std::mutex> guard(common->node_update_mutex);
-    rmw_dds_common::msg::ParticipantEntitiesInfo participant_msg =
-      common->graph_cache.add_node(common->gid, name, namespace_);
-    if (RMW_RET_OK != rmw_publish(
-        common->pub,
-        static_cast<void *>(&participant_msg),
-        nullptr))
-    {
-      // If publishing the message failed, we don't have to publish an update
-      // after removing it from the graph cache */
-      static_cast<void>(common->graph_cache.remove_node(common->gid, name, namespace_));
-      return nullptr;
-    }
+  // Though graph_cache methods are thread safe, both cache update and publishing have to also
+  // be atomic.
+  // If not, the following race condition is possible:
+  // node1-update-get-message / node2-update-get-message / node2-publish / node1-publish
+  // In that case, the last message published is not accurate.
```

Comment:
I think this comment doesn't belong here anymore, right?  That is, the comment is still correct, but the handling of that race is in `rmw_dds_common` now.